synthesizer/src/block/transaction/mod.rs

```diff
@@ -92,8 +96,13 @@ impl<N: Network> Transaction<N> {
         let deployment = vm.deploy(program, rng)?;
         // Compute the fee.
         let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_gates, query, rng)?;
+
```

Comment:
In line with the suggestion below:

```suggestion
        ensure!(!deployment.program().functions().is_empty(), "Attempted to create an empty transaction deployment");

```

Reply:
Added suggested change in ff8049abc